nemo/collections/common/parts/adapter_modules.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+
+from torch import nn as nn
+
+from nemo.collections.common.parts.utils import activation_registry
+
+
+class LinearAdapter(nn.Module):
+    """
+    Simple Linear Feedforward Adapter module with LayerNorm and singe hidden layer with activation function.
+    Note: The adapter explicitly initializes its final layer with all zeros in order to avoid affecting the
+    original model when all adapters are disabled.
+
+    Args:
+        in_features: Input dimension of the module. Note that for adapters, input_dim == output_dim.
+        dim: Hidden dimension of the feed forward network.
+        activation: Str name for an activation function.
+        norm_position: Str, can be `pre` or `post`. Defaults to `post`. Determines whether the normalization
+            will occur in the first layer or the last layer. Certain architectures may prefer one over the other.
+    """
+
+    def __init__(self, in_features, dim, activation: str = 'swish', norm_position="post"):
+        super().__init__()
+
+        activation = activation_registry[activation]()
+        # If the activation can be executed in place, do so.
+        if hasattr(activation, 'inplace'):
+            activation.inplace = True
+
+        assert norm_position in ['pre', 'post']
+        self.norm_position = norm_position
+
+        if norm_position == 'pre':
```

Comment:
One option here is to add all the layers to a list and then decide to add the normalization layer in the beginning or the end of the list. Then use the list to create the nn.Sequential.

Reply:
Right thats one option. This is cleaner for the time being from a statedict side so lets keep this. Im also a bit concerned by export so lets try this for now.